### PR TITLE
[wip] - Feature(point): Add basic support for built in postgres geometric point

### PR DESCRIPTION
**Pull request by jbavari**
This PR is a work in progress.

I've got the basic Postgres Geometric point type implemented.

There is a query added to `query_test.exs` to test encoding and decoding to postgrex/extensions/binary.ex. 

I've added in Point type to builtins.ex. 

The reason I'm filing this PR is to get some feedback and see if I am on the right path.

If this PR is looking good, I'll go ahead and add the other geometric types:

```
lseg
path
box
poly
line
circle
```

References: 
* [Postgres Geometric (built in) types](http://www.postgresql.org/docs/9.4/static/datatype-geometric.html)
* [Postgres Source code for Point Struct](https://github.com/postgres/postgres/blob/master/src/include/utils/geo_decls.h#L58)
* Query to get geometric types and their corresponding `oid` and `send` values: `select * from pg_type where typcategory = 'G';`

Please let me know if there is anything else you'd like me to correct here.

Thanks for taking the time to look at this!

### CODE REVIEW

File: lib/postgrex/builtins.ex [hunk 1]
```diff
@@ -146,3 +146,19 @@ defmodule Postgrex.MACADDR do
...
+
+defmodule Postgrex.Point do
+  @moduledoc """
+  Struct for Postgres point.
+
+  ## Fields
+    * `x`
+    * `y`
+  """
+  @type t :: %__MODULE__{x: :x, y: :x}
```
Comment (ericmj) [inline lib/postgrex/builtins.ex]: The types are wrong here, I think they should be `float`.

File: lib/postgrex/extensions/binary.ex [hunk 2]
```diff
@@ -670,6 +677,11 @@ defmodule Postgrex.Extensions.Binary do
   defp decode_network(<<a::8, b::8, c::8, d::8, e::8, f::8>>),
     do: %Postgrex.MACADDR{address: {a, b, c, d, e, f}}
 
+  # Geometrics
+  defp decode_point(<<x :: float64, y :: float64>>) do
+    %Postgrex.Point{ x: x, y: y }
```
Comment (ericmj) [inline lib/postgrex/extensions/binary.ex]: :scissors: extra spaces. `%Postgrex.Point{x: x, y: y}`

**Comment by ericmj** [comment]
Looks good, just left two comments. Please take a look at the test failures also: https://travis-ci.org/ericmj/postgrex/builds/95478227

File: lib/postgrex/builtins.ex [hunk 3]
```diff
@@ -146,3 +146,19 @@ defmodule Postgrex.MACADDR do
...
+
+  defstruct [
+    x: nil,
+    y: nil]
+
```
Comment (lexmag) [inline lib/postgrex/builtins.ex]: Needless line :scissors:.

**Comment by ericmj** [comment]
If you fix the failing test and rebase to a single commit this will be ready to merge.

**Comment by jbavari** [comment]
@ericmj I've updated the test to strictly use floats. The tests appear to pass, however, it looks like there was some error with one of the Postgres builds due to some `mix deps.get` call.

In my repo's Travis CI build: https://travis-ci.org/jbavari/postgrex/builds/101504381

Can we trigger the build again to see if that fixes the issue?

**Comment by ericmj** [comment]
It passes now, please squash into a single commit.

**Comment by InAbsentia** [comment]
Hi @jbavari @ericmj 

Thanks for all your work so far on Postgrex and adding the point type!

I could really use this feature for distance queries. I squashed, rebased from upstream/master, and updated to the new extension format in a fork so I have something to work with in my own app while this gets sorted out.

It's over here if you want to check it out and let me know how you want to proceed: https://github.com/InAbsentia/postgrex/commit/4eca058bb3c2509bd0fba8446e01b07930396a5b

I hope it helps get this wrapped up.

**Comment by ericmj** [comment]
Closing this PR since it's been dead for a month. If you have working, rebased code please open a new PR.

**Comment by jbavari** [comment]
@ericmj Sorry about the delay. I had rebased, but due to some merging issues I ran into, I got embarassed and felt defeated. I have it somewhat working, but I still do not have other geometric shapes supported (multi-polyline, polyline, etc).

@InAbsentia I'd love to continue helping on the feature if there's anything else you're working on.

**Comment by InAbsentia** [comment]
@ericmj Thanks. I'll open a new PR.

@jbavari Thanks for your candid response. It was a pretty rough rebase because of the upstream changes, so I figured you just didn't have time to deal with it and lost interest. I don't have a need for the other geometric types on my current project, but I'm willing to help you get the rest of them in if you're still interested. :smiley:

**Pull request closed without merging**